plugin/fuzz/source/dextool/plugin/fuzz/frontend/fuzz.d

```diff
@@ -0,0 +1,250 @@
+                base_filename ~ "_global" ~ implExt));
+
+        if(xml_dir.length == 0)
+            return;
+        if(compile_db.length == 0)
+            return;
+        this.xml_dir = BaseDir(xml_dir[0]);
+        this.compile_db = compile_db.fromArgCompileDb;
+    }
+
+    //Parameter functionis
+    @trusted string[] getIncludes() {
+        import std.algorithm : map;
+        import std.array;
+        import std.path : baseName;
+        import std.stdio;
+
+        string[] ret;
+
+        foreach(a ; this.compile_db.getHeaderFiles) {
+            writeln(a);
+            ret ~= a;
+        }
+
+        return ret; //this.compile_db.getHeaderFiles.map!(a => baseName(a)).array;
+
+
+    }
+    Parameters.Files getFiles() {
+        return Parameters.Files(main_file_hdr, main_file_impl,
+                main_file_globals);
+    }
+
+    MainName getMainName() {
+        return main_name;
+    }
+
+    MainNs getMainNs() {
+        return main_ns;
+    }
+
+    MainInterface getMainInterface() {
+        return main_if;
+    }
+
+    StubPrefix getArtifactPrefix() {
+        return prefix;
+    }
+
+    DextoolVersion getToolVersion() {
+        import dextool.utility : dextoolVersion;
+
+        return dextoolVersion;
+    }
+
+    CustomHeader getCustomHeader() {
+        return CustomHeader("");
+    }
+
+    BaseDir getXMLBasedir() {
+        return this.xml_dir;
+    }
+
+    CompileCommandDB getCompileDB() {
+        return this.compile_db;
+    }
+
+    //Product functions
+    void putFile(FileName fname, CppHModule hdr_data) {
+        file_data ~= FileData(fname, hdr_data.render());
+    }
+
+    void putFile(FileName fname, CppModule impl_data) {
+        file_data ~= FileData(fname, impl_data.render());
+    }
+}
+
+ExitStatusType genCpp(FuzzVariant variant) {
+    import std.conv : text;
+    import std.path : buildNormalizedPath, asAbsolutePath;
+    import std.typecons : Yes;
+
+    import cpptooling.analyzer.clang.context : ClangContext;
+    import cpptooling.data.representation : CppRoot;
+    import dextool.compilation_db : defaultCompilerFlagFilter;
+    import dextool.plugin.backend.fuzz.fuzzvariant : Generator,
+        FuzzVisitor;
+    import dextool.io : writeFileData;
+    import std.stdio;
+
+    auto visitor = new FuzzVisitor!(CppRoot, Products)(variant);
+    string[] use_cflags;
+    string[] in_cflags;
+    auto user_cflags = prependDefaultFlags(in_cflags, "-xc++");
+
+    auto hfiles = variant.getCompileDB.getHeaderFiles();
+    //auto range = variant.getCompileDB.map!(a => a.absoluteFile).enumerate;
+
+    string res;
+    writeln(hfiles);
+    //
+    foreach(hfile, cmd; hfiles) {
+        auto db_search_result = variant.getCompileDB.appendOrError(user_cflags, cmd[1],
+                CompileCommandFilter(defaultCompilerFlagFilter, 1));
+
+        use_cflags = db_search_result.get.cflags;
+
+        auto ctx = ClangContext(Yes.useInternalHeaders, Yes.prependParamSyntaxOnly);
```

Comment:
tip: the ClangContext can be moved outside of the loop. Make it a little bit more efficient.